src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java

```diff
@@ -40,12 +42,19 @@ public void runTests()
 
         //Configure Fast-RTPS for the tests
         ArrayList<String[]> commands = new ArrayList<String[]>();
-        commands.add(new String[]{"mkdir -p " + OUTPUT_PATH, "."});
-        commands.add(new String[]{"rm -rf Fast-RTPS", OUTPUT_PATH});
-        commands.add(new String[]{"git clone -b " + branch + " https://github.com/eProsima/Fast-RTPS.git", OUTPUT_PATH});
-        commands.add(new String[]{"mkdir build", OUTPUT_PATH + "/Fast-RTPS"});
+        if (Files.notExists(Paths.get(OUTPUT_PATH + "/Fast-RTPS/build")))
```

Comment:
What if the path exists but using a branch different from the one we currently want? Would we be using the old one?

Reply:
Take a look at the `else` branch.

Reply:
Sorry I completely missed that :(